File: app/src/main/java/com/app/dr1009/emptyrecyclerviewdemo/MainActivity.java
package com.app.dr1009.emptyrecyclerviewdemo;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;

import com.app.dr1009.emptyrecyclerview.EmptyRecyclerView;

import java.util.ArrayList;
import java.util.List;

import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {

    private final RecyclerAdapter mAdapter = new RecyclerAdapter();
    private final List<String> mData = new ArrayList<>();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        // init EmptyRecyclerView
        EmptyRecyclerView recyclerView = findViewById(R.id.recycler);
        recyclerView.setAdapter(mAdapter);

        // init Buttons
        Button addButton = findViewById(R.id.button_add);
        addButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                mData.add("Item No." + (mData.size() + 1));
                mAdapter.submitList(new ArrayList<>(mData));
            }
        });

        Button clearButton = findViewById(R.id.button_clear);
        clearButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                mData.clear();
                mAdapter.submitList(null);
            }
        });
    }
}

File: app/src/main/java/com/dr1009/app/emptyrecyclerviewdemo/MainActivity.kt
package com.dr1009.app.emptyrecyclerviewdemo

import android.os.Bundle
import android.widget.Button
import androidx.appcompat.app.AppCompatActivity
import com.dr1009.app.emptyrecyclerview.EmptyRecyclerView

class MainActivity : AppCompatActivity(R.layout.activity_main) {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        // init EmptyRecyclerView
        val recyclerView = findViewById<EmptyRecyclerView>(R.id.recycler)
        val adapter = RecyclerAdapter()
        recyclerView.adapter = adapter

        val data: MutableList<String> = mutableListOf()

        // init Buttons
        val addButton = findViewById<Button>(R.id.button_add)
        addButton.setOnClickListener {
            data.add("Item No." + (data.size + 1))
            adapter.submitList(data.toList())
        }
        val clearButton = findViewById<Button>(R.id.button_clear)
        clearButton.setOnClickListener {
            data.clear()
            adapter.submitList(data.toList())
        }
    }
}